### PR TITLE
[SystemZ][z/OS] Add UtcClock extension to chrono.h/.cpp

### PR DESCRIPTION
**Pull request by ysyeda**
This PR adds handling for UtcClock to chrono.h/.cpp.

### CODE REVIEW

**Comment by ysyeda** [comment]
ping

File: llvm/lib/Support/Chrono.cpp [hunk 1]
```diff
@@ -50,44 +68,63 @@ raw_ostream &operator<<(raw_ostream &OS, TimePoint<> TP) {
...
-  struct tm LT = getStructTM(Truncated);
   // Handle extensions first. strftime mangles unknown %x on some platforms.
-  if (Style.empty()) Style = "%Y-%m-%d %H:%M:%S.%N";
+  if (Style.empty())
+    Style = "%Y-%m-%d %H:%M:%S.%N";
   std::string Format;
   raw_string_ostream FStream(Format);
   for (unsigned I = 0; I < Style.size(); ++I) {
-    if (Style[I] == '%' && Style.size() > I + 1) switch (Style[I + 1]) {
-        case 'L':  // Milliseconds, from Ruby.
```
Comment (dwblaikie) [inline llvm/lib/Support/Chrono.cpp:65]: This and other parts seem to be reformatted without changes to the code? Could you remove the unrelated reformatting and/or commit it separately if there's a need?

Comment (ysyeda) [reply llvm/lib/Support/Chrono.cpp]: Thanks, the unrelated formatting is removed now.

**Comment by ysyeda** [comment]
ping

**Comment by dwblaikie** [comment]
Are there patches outstanding that'll use this in the LLVM project?

**Review by dwblaikie — APPROVED** [review]

**Comment by ysyeda** [comment]
> Are there patches outstanding that'll use this in the LLVM project?

@dwblaikie This patch is required for a change in the SystemZ backend for z/OS: https://github.com/llvm/llvm-project/pull/65407 I will post the PR once this one has been merged. Thanks for your review.
